pkg/cmd/pulumi/login.go

```diff
@@ -120,6 +120,10 @@ func newLoginCmd() *cobra.Command {
 				if err != nil {
 					return fmt.Errorf("could not determine current cloud: %w", err)
 				}
+			} else if url := strings.TrimPrefix(strings.TrimPrefix(
+				cloudURL, "https://"), "http://"); strings.HasPrefix(url, "app.pulumi.com/") {
+				return fmt.Errorf("%s is not a valid self-hosted backend, "+
+					"use `pulumi login` without any arguments to log into the Pulumi service backend", cloudURL)
```

Comment:
this is kinda nity, so feel free to ignore if you feel strongly

```suggestion
					"use only `pulumi login` to log into the Pulumi service backend", cloudURL)
```